vmm/src/vstate.rs

```diff
@@ -275,13 +290,139 @@ impl Vcpu {
         Ok(())
     }
 
-    /// Runs the VCPU until it exits, returning the reason.
+    /// Main loop of the vCPU thread.
+    ///
     ///
+    /// Runs the vCPU in KVM context in a loop. Handles KVM_EXITs then goes back in.
+    /// Also registers a signal handler to be able to kick this thread out of KVM_RUN.
     /// Note that the state of the VCPU and associated VM must be setup first for this to do
     /// anything useful.
-    pub fn run(&self) -> Result<VcpuExit> {
+    pub fn run(
+        &mut self,
+        thread_barrier: Arc<Barrier>,
+        seccomp_level: u32,
+        io_bus: devices::Bus,
+        mmio_bus: devices::Bus,
+        vcpu_exit_evt: EventFd,
+    ) {
+        unsafe {
```

Comment:
Like you also rightfully mentioned, this part of the code is written as if we're setting the signal handler per thread, when it's actually set for the entire process. It looks like we can install the handler in the Vmm before running the vcpus.

Reply:
Yes, I have that change in a following PR, but I agree it might be better suited in this one.

Reply:
done